### PR TITLE
Expose docker-compose up --build flag

### PR DESCRIPTION
**Pull request by dylanscott**
TL;DR: This PR adds a parameter to `Docker::Compose::Session`'s `up` method to pass the `--build` flag to `docker-compose`.

Long version: There seems to be some long and confusing (at least to me) history with these flags, and despite their names they are not inverses of one another. It appears that way back in the day (when `docker-compose` was `fig`) the default behavior was to build images before starting containers, so a `--no-build` flag [was added](https://github.com/docker/compose/pull/586) to speed up the command by suppressing this behavior. Later, [the default behavior was changed](https://github.com/docker/compose/issues/693) and the `--build` flag [was added](https://github.com/docker/compose/pull/2601) to force a build.

### CODE REVIEW

**Comment by xeger** [comment]
Hah; I had this exact same change queued up in my local master, but forgot about committing it!

**Comment by xeger** [comment]
To add to the confusion @dylanscott, various minor versions of Docker Compose have temporarily removed one or the other of these flags. Before I added the "don't specify defaulted options" logic -- ugly though it is in our Ruby codebase -- this was a huge problem. With the non-defaulting in place, we can support the union of all CLI parameters that any version of Compose has ever supported.

I'm going to publish v1.0.0 with your changes. I've been using the pre-release for several weeks now and haven't encountered any troubles; this was the last item on my list. Thanks for the contribution.

**Comment by dylanscott** [comment]
Awesome, thanks for the quick turnaround!
